zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateEscalationEventTest.java

```diff
@@ -0,0 +1,540 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.processinstance.migration;
+
+import static io.camunda.zeebe.engine.processing.processinstance.migration.MigrationTestUtil.extractProcessDefinitionKeyByProcessId;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.protocol.record.Assertions;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.protocol.record.value.BpmnElementType;
+import io.camunda.zeebe.test.util.BrokerClassRuleHelper;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class MigrateEscalationEventTest {
+
+  private static final String ESCALATION_CODE_NUMBER = "404";
+  private static final String ESCALATION_CODE_NUMBER_2 = "405";
+  private static final String ESCALATION_CODE_NUMBER_3 = "406";
+  private static final String ESCALATION_CODE_NUMBER_4 = "407";
+  @Rule public final EngineRule engine = EngineRule.singlePartition();
```

Comment:
🔧 We can reuse the engine between the test cases to speed up these tests.

```suggestion
  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
```